sandbox/plc4c/examples/hello-subsciptions/src/hello_subscriptions.c

```diff
@@ -0,0 +1,264 @@
+  result = plc4c_system_add_driver(system, simulated_driver);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  printf("Registering transport for the 'dummy' transport ... ");
+  plc4c_transport *dummy_transport = plc4c_transport_dummy_create();
+  result = plc4c_system_add_transport(system, dummy_transport);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  // Initialize the plc4c_system (loading of drivers, setting up other stuff,
+  // ...)
+  printf(
+      "Initializing the PLC4C system (Loading of drivers and transports) ... ");
+  result = plc4c_system_init(system);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+  printf("SUCCESS\n");
+
+  // Establish connections to remote devices
+  // you may or may not care about the connection handle
+  printf("Connecting to 'simulated://foo' ... ");
+  result = plc4c_system_connect(system, "simulated://foo", &connection);
+  if (result != OK) {
+    printf("FAILED\n");
+    return -1;
+  }
+
+  // Central program loop ...
+  plc4c_connection_state state = CONNECTING;
+  while (loop) {
+    printf("* ");
+
+    // Give plc4c a chance to do something.
+    // This is where all I/O is being done.
+    if (plc4c_system_loop(system) != OK) {
+      printf("ERROR in the system loop\n");
+      break;
+    }
+
+    // Depending on the current state, implement some logic.
+    switch (state) {
+      case CONNECTING: {
+        // Check if the connection is established:
+        if (plc4c_connection_get_connected(connection)) {
+          printf("SUCCESS\n");
+          state = CONNECTED;
+        } else if (plc4c_connection_has_error(connection)) {
+          printf("FAILED\n");
+          return -1;
+        }
+        break;
+      }
+      case CONNECTED: {
+        // Create a new read-request.
+        printf("Preparing a read-request ... ");
+
+        result = plc4c_connection_create_subscription_request(
+            connection, &subscription_request);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        }
+        printf("SUCCESS\n");
+
+        printf("Adding cyclic item for 'RANDOM/foo:INTEGER' ... ");
+        result = plc4c_subscription_request_add_cyclic_item(
+            subscription_request, "RANDOM/foo:INTEGER", 500);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        }
+        printf("SUCCESS\n");
+
+        // Execute the read-request.
+        printf("Executing a read-request ... ");
+        result = plc4c_subscription_request_execute(
+            subscription_request, &subscription_request_execution);
+        if (result != OK) {
+          printf("FAILED\n");
+          return -1;
+        } else {
+          state = SUBSCRIPTION_REQUEST_SENT;
+        }
+        break;
+      }
+        // Wait until the read-request execution is finished.
```

Comment:
This makes it seem like the request being sent is equal to the response being received
I think the state would be SUBSCRIPTION_WAITING_RESPONSE